lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -205,7 +205,7 @@ DeclKind firrtl::getDeclarationKind(Value val) {
 }
 
 size_t firrtl::getNumPorts(Operation *op) {
-  if (auto module = dyn_cast<hw::HWModuleLike>(*op))
```

Comment:
I'm curious why this was necessary.

Reply:
Making this change is not necessary, this is just a cleanup opportunity I found while looking through all our casts to HWModuleLike.